Matrix3_TEST.py and  StopWatch_TEST.py tests fail on Windows with conda-forge dependencies

## Environment
* OS Version: Windows 10
* Source or binary build? Source, https://github.com/ignitionrobotics/ign-math/commit/ceab8c7f32f43c3ebf576146e88677ada17edf23, 6.10.0 release 

This problem was detected while enabling Python support in the libignition-math6 package on conda-forge, see https://github.com/conda-forge/libignition-math4-feedstock/pull/25/ .

<details>
  

`Matrix3_TEST.py`: 
~~~
2022-05-01T11:43:26.6555205Z 115/142 Test #115: Matrix3_TEST.py ........................***Failed    0.17 sec
2022-05-01T11:43:26.6570710Z ..........F..
2022-05-01T11:43:26.6571776Z ======================================================================
2022-05-01T11:43:26.6572650Z FAIL: test_to_quaternion (__main__.TestMatrix3)
2022-05-01T11:43:26.6573622Z ----------------------------------------------------------------------
2022-05-01T11:43:26.6574494Z Traceback (most recent call last):
2022-05-01T11:43:26.6575464Z   File "%SRC_DIR%\src\python_pybind11\test\Matrix3_TEST.py", line 310, in test_to_quaternion
2022-05-01T11:43:26.6576958Z     self.assertTrue(q == quatFromMat)
2022-05-01T11:43:26.6577872Z AssertionError: False is not true
2022-05-01T11:43:26.6578264Z 
2022-05-01T11:43:26.6578931Z ----------------------------------------------------------------------
2022-05-01T11:43:26.6579667Z Ran 13 tests in 0.004s
2022-05-01T11:43:26.6580156Z 
2022-05-01T11:43:26.6580693Z FAILED (failures=1)
2022-05-01T11:43:26.6581035Z 
~~~

`StopWatch_TEST.py`:
~~~
2022-05-01T11:43:40.4890605Z 131/142 Test #131: StopWatch_TEST.py ......................***Failed   11.17 sec
2022-05-01T11:43:40.4892371Z .F.F
2022-05-01T11:43:40.4893342Z ======================================================================
2022-05-01T11:43:40.4894340Z FAIL: test_equal_operator (__main__.TestBox)
2022-05-01T11:43:40.4895380Z ----------------------------------------------------------------------
2022-05-01T11:43:40.4896347Z Traceback (most recent call last):
2022-05-01T11:43:40.4897542Z   File "%SRC_DIR%\src\python_pybind11\test\StopWatch_TEST.py", line 101, in test_equal_operator
2022-05-01T11:43:40.4898597Z     self.runTimer(watch2)
2022-05-01T11:43:40.4899537Z   File "%SRC_DIR%\src\python_pybind11\test\StopWatch_TEST.py", line 39, in runTimer
2022-05-01T11:43:40.4900831Z     self.assertGreaterEqual(_time.elapsed_run_time() + handleSteadyClock, timedelta(seconds=1))
2022-05-01T11:43:40.4902387Z AssertionError: datetime.timedelta(microseconds=998771) not greater than or equal to datetime.timedelta(seconds=1)
2022-05-01T11:43:40.4903290Z 
2022-05-01T11:43:40.4904003Z ======================================================================
2022-05-01T11:43:40.4905073Z FAIL: test_start_stop_reset (__main__.TestBox)
2022-05-01T11:43:40.4906067Z ----------------------------------------------------------------------
2022-05-01T11:43:40.4907083Z Traceback (most recent call last):
2022-05-01T11:43:40.4908205Z   File "%SRC_DIR%\src\python_pybind11\test\StopWatch_TEST.py", line 125, in test_start_stop_reset
2022-05-01T11:43:40.4909302Z     self.runTimer(watch)
2022-05-01T11:43:40.4910333Z   File "%SRC_DIR%\src\python_pybind11\test\StopWatch_TEST.py", line 39, in runTimer
2022-05-01T11:43:40.4911701Z     self.assertGreaterEqual(_time.elapsed_run_time() + handleSteadyClock, timedelta(seconds=1))
2022-05-01T11:43:40.4913207Z AssertionError: datetime.timedelta(microseconds=999953) not greater than or equal to datetime.timedelta(seconds=1)
2022-05-01T11:43:40.4915090Z 
2022-05-01T11:43:40.4916809Z ----------------------------------------------------------------------
2022-05-01T11:43:40.4917673Z Ran 4 tests in 11.013s
2022-05-01T11:43:40.4918078Z 
2022-05-01T11:43:40.4918640Z FAILED (failures=2)
2022-05-01T11:43:40.4918985Z 
~~~


</details>

## Description
* Expected behavior: The test suite works fine,
* Actual behavior: The aforementioned test fail.

## Steps to reproduce

1. Create a conda-forge environment like `mamba create -n ignmathdev libignition-cmake2 eigen3 python pybind11 pybind11-abi vs2019_win-64 cmake ninja pkg-config`
2. Compile ign-cmake and run tests
3. Check the output of the tests

## Output
See log.

--- a/src/python_pybind11/test/Matrix3_TEST.py
+++ b/src/python_pybind11/test/Matrix3_TEST.py
@@ -307,7 +307,9 @@ class TestMatrix3(unittest.TestCase):
         q = Quaterniond(math.pi/2.0, math.pi/2.0, 0)
         matFromQuat = Matrix3d(q)
         quatFromMat = Quaterniond(matFromQuat)
-        self.assertTrue(q == quatFromMat)
+        # We check both cases as quaternion double-cover the rotation matrix
+        # space, see https://github.com/gazebosim/gz-math/issues/416#issuecomment-1741741183
+        self.assertTrue(q == quatFromMat or q == -quatFromMat)
 
         # test the cases where matrix trace is negative
         # (requires special handling)

